### PR TITLE
Added new GDScript language

### PR DESCRIPTION
**Pull request by khairul169**
Add support for gdscript language highlighting. GDScript is a programming language used in Godot Game Engine.

### CODE REVIEW

File: src/languages/gdscript.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			'Dictionary Array PoolByteArray PoolIntArray ' +
+			'PoolRealArray PoolStringArray PoolVector2Array ' +
+			'PoolVector3Array PoolColorArray',
+		
+		literal:
+			'true false null'
+	};
+	
+	return {
+		aliases: ['gd'],
```
Comment (ldct) [inline src/languages/gdscript.js]: can we avoid using such a short alias? I am worried about collisions with future languages.

Comment (khairul169) [reply src/languages/gdscript.js]: sure, i'll change that

Comment (TheCire) [reply src/languages/gdscript.js]: I'd suggest adding 'gdscript' also
```suggestion
		aliases: ['godot', 'gdscript'],
```

Comment (khairul169) [reply src/languages/gdscript.js]: Updated :)

**Comment by Jose-Moreno** [comment]
Any update on this? Would be great to see GDScript highlighting added :+1: :slightly_smiling_face:

**Comment by polarlighthouse** [comment]
Could you just solve the conflicts so it can be merged? :)

**Comment by marcoscaceres** [comment]
We need to move this out to its own repository. Would someone be willing to do that and maintain it?

**Comment by form-follows-function** [comment]
> We need to move this out to its own repository. Would someone be willing to do that and maintain it?

Since the author is not replying I could actually maintain this, as I use the language a lot on my blog. Will look into this.

**Comment by form-follows-function** [comment]
@marcoscaceres can you tell me what do you specifically mean by moving it out own from its own repository?

**Comment by marcoscaceres** [comment]
Hi @form-follows-function, at a minimum, it means setting up a small repo like this:
https://github.com/highlightjs/highlightjs-solidity

It just includes the language definition, how to include it, and package.json to publish the language definition to npm.

**Comment by form-follows-function** [comment]
@marcoscaceres I created the repository. Can I simply copy the original author's author or does it have to comply with the mentioned repository's .js conventions?

**Comment by marcoscaceres** [comment]
Sorry for the delay @form-follows-function. You can just copy it.

**Comment by form-follows-function** [comment]
@marcoscaceres No problem. I added all necessary files to the repository: 
[https://github.com/form-follows-function/highlightjs-gdscript](https://github.com/form-follows-function/highlightjs-gdscript)

Can you check if complies with the requirements?

**Comment by marcoscaceres** [comment]
@form-follows-function looking great. However, just needs  the following, which you can copy from terraform (just replace "Terraform" with "GDScript" as needed):

- [ ] var module shim...
https://github.com/highlightjs/highlightjs-terraform/blob/master/terraform.js#L16

- [ ] give the defining function a name
https://github.com/highlightjs/highlightjs-terraform/blob/master/terraform.js#L18

- [ ] Provide an export:
https://github.com/highlightjs/highlightjs-terraform/blob/master/terraform.js#L83-L87

**Comment by form-follows-function** [comment]
Thanks for pointing that out. I added these to the code

**Comment by marcoscaceres** [comment]
@form-follows-function that's great. Ok, last things: 

1. add me (@marcoscaceres) as a Contributor via the Repo's Settings. 
2. Go "Options" -> scroll down to "Danger Zone" -> "Transfer ownership", and transfer ownership to the "highlightjs" organization. 

Once that's done, I'll add you back as admin. 🍻

**Comment by form-follows-function** [comment]
@marcoscaceres Ive invited you to the repository but i can’t transfer it yet because it says i haven’t permission to add repositories to highlightjs

**Comment by marcoscaceres** [comment]
@form-follows-function whoopsy, I'd thought I'd granted you permission to create repos. I've done that now.

**Comment by form-follows-function** [comment]
Ok, now it got transferred 🙂

**Comment by marcoscaceres** [comment]
Ok, closing as done. We can pick up new issues etc. In the new repo. Thanks for your help @form-follows-function!

**Pull request closed without merging**